From: Stefan Hajnoczi <stefanha@gmail.com>
Date: Sun, 11 Nov 2012 13:24:44 +0100
Subject: [049/106] staging: line6: replace deprecated strict_strtol() in
 toneport.c

commit b07d945227e4e62d93e26c694bf0e7b70978a7d6 upstream.

The LED value is an int, so replace strict_strtol() with kstrtoint().
It's safe to pass in the actual variable instead of a local temporary
because strto*() doesn't write to the result unless the function returns
success.

Signed-off-by: Stefan Hajnoczi <stefanha@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/line6/toneport.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

Index: linux-3.2.46/drivers/staging/line6/toneport.c
===================================================================
--- linux-3.2.46.orig/drivers/staging/line6/toneport.c	2013-07-26 19:28:31.000000000 +0000
+++ linux-3.2.46/drivers/staging/line6/toneport.c	2013-07-26 19:29:02.000000000 +0000
@@ -127,13 +127,11 @@
 				    const char *buf, size_t count)
 {
 	int retval;
-	long value;
 
-	retval = strict_strtol(buf, 10, &value);
+	retval = kstrtoint(buf, 10, &led_red);
 	if (retval)
 		return retval;
 
-	led_red = value;
 	toneport_update_led(dev);
 	return count;
 }
@@ -143,13 +141,11 @@
 				      const char *buf, size_t count)
 {
 	int retval;
-	long value;
 
-	retval = strict_strtol(buf, 10, &value);
+	retval = kstrtoint(buf, 10, &led_green);
 	if (retval)
 		return retval;
 
-	led_green = value;
 	toneport_update_led(dev);
 	return count;
 }
